src/algorithms/model.ts

```diff
@@ -0,0 +1,376 @@
+    isolatedFrac: {},
+    importsPerDay: { total: params.importsPerDay },
+    timeDeltaDays,
+    // Dummy infectionRate function. This is set below.
+    infectionRate: () => -Infinity,
+    timeDelta: msPerDay * timeDeltaDays,
+    incubationTime: params.incubationTime,
+    populationServed: params.populationServed,
+    numberStochasticRuns: params.numberStochasticRuns,
+    hospitalBeds: params.hospitalBeds,
+    ICUBeds: params.ICUBeds,
+  }
+
+  // Compute age-stratified parameters
+  const total = severity.map(d => d.ageGroup).reduce((a, b) => a + ageCounts[b], 0)
+
+  let hospitalizedFrac = 0
+  let criticalFracHospitalized = 0
+  let fatalFracCritical = 0
+  let avgIsolatedFrac = 0
+  severity.forEach(d => {
+    const freq = (1.0 * ageCounts[d.ageGroup]) / total
+    pop.ageDistribution[d.ageGroup] = freq
+    pop.infectionSeverityRatio[d.ageGroup] = (d.severe / 100) * (d.confirmed / 100)
+    pop.infectionCritical[d.ageGroup] = pop.infectionSeverityRatio[d.ageGroup] * (d.critical / 100)
+    pop.infectionFatality[d.ageGroup] = pop.infectionCritical[d.ageGroup] * (d.fatal / 100)
+
+    const dHospital = pop.infectionSeverityRatio[d.ageGroup]
+    const dCritical = d.critical / 100
+    const dFatal = d.fatal / 100
+
+    // d.isolated is possible undefined
+    const isolated = d.isolated ?? 0
+    hospitalizedFrac += freq * dHospital
+    criticalFracHospitalized += freq * dCritical
+    fatalFracCritical += freq * dFatal
+    avgIsolatedFrac += (freq * isolated) / 100
+
+    // Age specific rates
+    pop.isolatedFrac[d.ageGroup] = isolated / 100
+    pop.recoveryRate[d.ageGroup] = (1 - dHospital) / params.infectiousPeriod
+    pop.hospitalizedRate[d.ageGroup] = dHospital / params.infectiousPeriod
+    pop.dischargeRate[d.ageGroup] = (1 - dCritical) / params.lengthHospitalStay
+    pop.criticalRate[d.ageGroup] = dCritical / params.lengthHospitalStay
+    pop.stabilizationRate[d.ageGroup] = (1 - dFatal) / params.lengthICUStay
+    pop.deathRate[d.ageGroup] = dFatal / params.lengthICUStay
+    pop.overflowDeathRate[d.ageGroup] = params.overflowSeverity * pop.deathRate[d.ageGroup]
+  })
+
+  // Get import rates per age class (assume flat)
+  const L = Object.keys(pop.recoveryRate).length
+  Object.keys(pop.recoveryRate).forEach(k => {
+    pop.importsPerDay[k] = params.importsPerDay / L
+  })
+
+  // Population average rates
+  pop.recoveryRate.total = (1 - hospitalizedFrac) / params.infectiousPeriod
+  pop.hospitalizedRate.total = hospitalizedFrac / params.infectiousPeriod
+  pop.dischargeRate.total = (1 - criticalFracHospitalized) / params.lengthHospitalStay
+  pop.criticalRate.total = criticalFracHospitalized / params.lengthHospitalStay
+  pop.deathRate.total = fatalFracCritical / params.lengthICUStay
+  pop.stabilizationRate.total = (1 - fatalFracCritical) / params.lengthICUStay
+  pop.overflowDeathRate.total = (params.overflowSeverity * fatalFracCritical) / params.lengthICUStay
+  pop.isolatedFrac.total = avgIsolatedFrac
```

Comment:
Previously, `pop` was being initialized to `{ ...params }` so it had access to all of fields that `params` had. Now that the initialization of `pop` has changed, I had to grab a lot of information from `params` instead.